nomenclature/testing.py

```diff
@@ -36,11 +36,14 @@ def assert_valid_structure(path: Path):
         :class:`DataStructureDefinition`
       - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
     """
-    definition = nomenclature.DataStructureDefinition(path / "definitions")
-    if (path / "mappings").is_dir():
+
+    definition = nomenclature.DataStructureDefinition(path / str(definitions))
+    if (path / str(mappings)).is_dir():
```

Comment:
I think the `str()` conversion is not strictly needed
```suggestion
    if (path / mappings).is_dir():
```